### PR TITLE
Dev a

### PR DESCRIPTION
**Pull request by AshutoshSancheti**
Hi Chinmay,

I have added functions to calculate Net present value of a stock using DCF & SSGR. Please review them on how I calculate.

### CODE REVIEW

File: lib/analyser.py [hunk 1]
```diff
@@ -3,3 +3,134 @@
...
+        #Calculate the discounted value of cash after capex spending
+        discounted_profit = (profit-capex)/(pow((1+discount_rate),(year+1)))
+        net_present_value += discounted_profit
+
+    terminal_value = profit*(1+perpetual_growth_rate)/(discount_rate-perpetual_growth_rate)
+    discounted_terminal_value = terminal_value/(pow((1+discount_rate),(year+1)))
+    net_present_value += discounted_terminal_value
+    return (market_cap/net_present_value)
+
+def DCF_MC_analysis(self, init_profit, init_fixed_asset, market_cap):
```
Comment (ccpandhare) [inline lib/analyser.py]:
```suggestion
def dcf_mc_analysis(self, init_profit, init_fixed_asset, market_cap):
```

File: lib/analyser.py [hunk 2]
```diff
@@ -3,3 +3,134 @@
...
+    # TODO : Try to add a dataframe based operation here to check if it is faster
+    for yud in years_until_decay:
+        for pgr in perpetual_growth_rate:
+            for j in range(iterations):
+                npv_distribution.append(self.calc_NPV(init_profit, init_profit_growth_rate[j], \
+                pgr, yud, discount_rate[j], init_fixed_asset, \
+                market_cap, no_of_years)) 
+    return np.mean(npv_distribution)+3*np.std(npv_distribution)
+
+def Calc_SSGR_revenue(revenue, NPM, DPR, fixed_assets_start, depreciation):
```
Comment (ccpandhare) [inline lib/analyser.py]:
```suggestion
def calc_ssgr_revenue(revenue, npm, dpr, fixed_assets_start, depreciation):
```

File: lib/analyser.py [hunk 3]
```diff
@@ -3,3 +3,134 @@
...
+    # Fixed assets at the beginning of the next year assuming all the funds available are reinvested
+    # Assumed deprec
+    fixed_assets_end = (fixed_assets_start*(1 - (depreciation/fixed_assets_start))) + funds_available
+    # Expected sales
+    expected_sales = fixed_assets_end*(revenue/fixed_assets_start)
+    SSGR = (expected_sales/revenue) - 1
+    return SSGR
+    #return (((((fixed_assets_start*(1 - (depreciation/fixed_assets_start))) + (revenue*NPM*(1-DPR)))*NFAT)/revenue) - 1)
+
+def Calc_SSGR_CFO(CFO, PAT, DPR, fixed_assets_start, depreciation):
```
Comment (ccpandhare) [inline lib/analyser.py]:
```suggestion
def calc_ssgr_cfo(cfo, pat, dpr, fixed_assets_start, depreciation):
```

**Review by ccpandhare — CHANGES_REQUESTED** [review]
Hey @AshutoshSancheti! Code looks good to me! It's a bit hard to review math in python - have you tested this on real data to make sure it's okay?

I've left a few naming suggestions - The standard convention is that only class names begin with capital letters :D If you look at the github syntax highlighting closely it treats identifiers starting with capital letters differently too (Please change the function bodies too)

File: lib/analyser.py (hunk 1 above)
Comment (AshutoshSancheti) [reply lib/analyser.py]: Followed the standard convention and implemented the changes

File: lib/analyser.py (hunk 2 above)
Comment (AshutoshSancheti) [reply lib/analyser.py]: Followed the standard convention and implemented the changes

File: lib/analyser.py (hunk 3 above)
Comment (AshutoshSancheti) [reply lib/analyser.py]: Followed the standard convention and implemented the changes

**Comment by AshutoshSancheti** [comment]
Regarding the math, I have tested the code out for the math and the distributions. The assumption here is that it our provide the metric for a given year's datapoints only. This is good but in the future there is a possibility that we might need to calculate the ssgr metrics for a range of years rather than the latest year and may have to modify the function to use a pd.Series or pd.dataframe. I don't want to deliberate on that decision now as I don't have enough inputs. 
Bottomline: functions work

**Comment by ccpandhare** [comment]
@AshutoshSancheti 
re: Math - makes sense!

fyi - looks like you’ve missed one function Calc_SSGR_CFO(CFO, …) -> please rename this function and it’s parameters before you merge 

thanks!

**Comment by AshutoshSancheti** [comment]
Forgot to save those changes. Committed and Pushed.

**Pull request merged**